azblob/list.go

```diff
@@ -12,41 +12,80 @@ import (
 )
 
 // Count counts the number of blobs filtered by the given tags filter
-func (azp *Storer) Count(ctx context.Context, tagsFilter string) (int64, error) {
+func (azp *Storer) Count(ctx context.Context, tagsFilter string, opts ...Option) (int64, error) {
 
 	logger.Sugar.Debugf("Count")
 
-	blobs, err := azp.FilteredList(ctx, tagsFilter)
+	r, err := azp.FilteredList(ctx, tagsFilter, opts...)
 	if err != nil {
 		return 0, err
 	}
 
-	return int64(len(blobs)), nil
+	return int64(len(r.Items)), nil
+}
+
+type FilterResponse struct {
+	Marker ListMarker // nil if no more pages
+
+	// Standard request status things
+	StatusCode int // For If- header fails, err can be nil and code can be 304
+	Status     string
+
+	Items []*azStorageBlob.FilterBlobItem
 }
 
 // FilteredList returns a list of blobs filtered on their tag values.
 //
-// tagsFilter example: "dog='germanshepherd' and penguin='emperorpenguin'"
-// Returns all blobs with the specific tag filter
-func (azp *Storer) FilteredList(ctx context.Context, tagsFilter string) ([]*azStorageBlob.FilterBlobItem, error) {
-	logger.Sugar.Debugf("FilteredList")
+// tagsFilter examples:
+//
+//	All blobs in a storage account
+//		"cat='tiger' AND penguin='emperorpenguin'"
+//	All blobs in a specific container
+//		"@container='zoo' AND cat='tiger' AND penguin='emperorpenguin'"
```

Comment:
nice, are there any examples of > time or > number?

Reply:
yeah, in the test in forestrie. I'll add them to the comments here tho!